libraries/RC_Channel/RC_Channel.h

```diff
@@ -213,6 +213,7 @@ class RC_Channel {
         // options 150-199 continue user rc switch options
         CRUISE =             150,  ///CRUISE mode
         TURTLE =             151,  // Turtle mode - flip over after crash
+        ARMDISARM_WITH_AIRMODE = 152, // arm or disarm vehicle enableing airmode
```

Comment:
To not change existing behaviour we must rename the old number to `ARMDISARM_WITH_AIRMODE` and create a new `ARMDISARM` here

Reply:
I think changing the existing behavior is the correct thing to do. 

Getting airmode when you don't want it can cause unexpected takeoffs and make disarming hard, generally quite nasty things that are not easy to fix, you cant just drop throttle to 0, would would have to have a go switch your disarm switch.

Not having airmode when you expected it causes you to go "oops" and instinctively raise throttle at which point stabilization is active again. 

We will have to add it to release notes so that those who wanted airmode on arming can get it back again.

Reply:
> I think changing the existing behavior is the correct thing to do.

Not convinced :-)

> Getting airmode when you don't want it can cause unexpected takeoffs and make disarming hard, generally quite nasty things that are not easy to fix, you cant just drop throttle to 0, would would have to have a go switch your disarm switch.

Unexpectedly not having airmode because you upgraded your firmware is likely to cause unexpected crashes :-)

> Not having airmode when you expected it causes you to go "oops" and instinctively raise throttle at which point stabilization is active again.
> 
> We will have to add it to release notes so that those who wanted airmode on arming can get it back again.

Still not getting this.  Users that already have airmode on are used to the behaviour they have.  They may choose to switch to the "no airmode" option - if they read the release notes and realise it exists.  New users would have the option of choosing one behaviour or the other according to their preferences/guesses.

I just don't see any advantages in not taking the conservative upgrade path.

Reply:
> Unexpectedly not having airmode because you upgraded your firmware is likely to cause unexpected crashes :-)

I disagree, this only changes acro and stabilize, in which case your actively flying the vehicle and spot it is not doing what you expect and instinctively raise the throttle avoiding the crash. 

> Still not getting this. Users that already have airmode on are used to the behaviour they have. They may choose to switch to the "no airmode" option - if they read the release notes and realise it exists. New users would have the option of choosing one behaviour or the other according to their preferences/guesses.

I'm thinking of the users that have no idea what airmode is and just currently have it by accident, for them they don't know that they have airmode so cant take that decision to turn it off. 

TLDR: I think having airmode when you didn't want it is worse than not having airmode when you did.

Reply:
> TLDR: I think having airmode when you didn't want it is worse than not having airmode when you did.

100% agree with this, as somebody who has been using airmode for many years.

Airmode can cause unexpected behaviour, which can result in damage or injury. It should be a conscious, explicit decision to activate it, it shouldn't be bundled as a side-effect of activating something else. I for one didn't even realise until too late that switch arming would also enable airmode, which still seems like a poor decision that should now be addressed with this commit.